packages/ui/components/ui/tx/actions-views/ibc-relay.tsx

```diff
@@ -4,93 +4,119 @@ import {
   FungibleTokenPacketData,
   IbcRelay,
 } from '@penumbra-zone/protobuf/penumbra/core/component/ibc/v1/ibc_pb';
-import { MsgRecvPacket } from '@penumbra-zone/protobuf/ibc/core/channel/v1/tx_pb';
+import {
+  MsgAcknowledgement,
+  MsgRecvPacket,
+  MsgTimeout,
+  MsgTimeoutOnClose,
+} from '@penumbra-zone/protobuf/ibc/core/channel/v1/tx_pb';
 import { MsgUpdateClient } from '@penumbra-zone/protobuf/ibc/core/client/v1/tx_pb';
 import { UnimplementedView } from './unimplemented-view.tsx';
 import { uint8ArrayToBase64 } from '@penumbra-zone/types/base64';
+import { ReactElement } from 'react';
+import { Packet } from '@penumbra-zone/protobuf/ibc/core/channel/v1/channel_pb';
 import { getUtcTime } from './isc20-withdrawal.tsx';
-import { useMemo } from 'react';
 
-// Packet data stored as json string encoded into bytes
-const parsePacket = ({ packet }: MsgRecvPacket): FungibleTokenPacketData | undefined => {
-  if (!packet?.data) {
+// Attempt to parse data w/ fallbacks if unknown or failures
+const ParsedPacketData = ({
+  data,
+  dataParser,
+}: {
+  data: Uint8Array;
+  dataParser?: (arg: Uint8Array) => ReactElement;
+}) => {
+  if (!data.length) {
     return undefined;
   }
 
+  if (!dataParser) {
+    return <ActionDetails.Row label='Data'>Unknown packet data</ActionDetails.Row>;
+  }
+
   try {
-    const dataString = new TextDecoder().decode(packet.data);
-    return FungibleTokenPacketData.fromJsonString(dataString);
+    return dataParser(data);
   } catch (e) {
-    return undefined;
+    return <ActionDetails.Row label='Data'>Error while parsing data</ActionDetails.Row>;
   }
 };
 
-const MsgResvComponent = ({ packet }: { packet: MsgRecvPacket }) => {
-  const packetData = useMemo(() => parsePacket(packet), [packet]);
+const PacketRows = ({
+  packet,
+  dataParser,
+}: {
+  packet: Packet;
+  dataParser?: (arg: Uint8Array) => ReactElement;
+}) => {
+  return (
+    <>
+      <ParsedPacketData data={packet.data} dataParser={dataParser} />
+      {!!packet.sequence && (
```

Comment:
comment: what does a sequencer number of 0 represent?

Reply:
Good catch, will change any field that returns a primitive (required field) to show without a conditional check.